wardrobe/money/src/lib.rs

```diff
@@ -84,87 +68,148 @@ pub enum ConstraintCheckerError {
-                    ConstraintCheckerError::OutputsExceedInputs
-                );
-
-                // Priority is based on how many token are burned
-                // Type stuff is kinda ugly. Maybe division would be better?
-                let burned = total_input_value - total_output_value;
-                Ok(if burned < u64::max_value() as u128 {
-                    burned as u64
-                } else {
-                    u64::max_value()
-                })
-            }
-            Self::Mint => {
-                // Make sure there are no inputs being consumed
-                ensure!(
-                    input_data.is_empty(),
-                    ConstraintCheckerError::MintingWithInputs
-                );
-
-                // Make sure there is at least one output being minted
-                ensure!(
-                    !output_data.is_empty(),
-                    ConstraintCheckerError::MintingNothing
-                );
-
-                // Make sure the outputs are the right type
-                for utxo in output_data {
-                    let utxo_value = utxo
-                        .extract::<Coin<ID>>()
-                        .map_err(|_| ConstraintCheckerError::BadlyTyped)?
-                        .0;
-                    ensure!(utxo_value > 0, ConstraintCheckerError::ZeroValueCoin);
-                }
-
-                // No priority for minting
-                Ok(0)
-            }
+    ) -> Result<ConstraintCheckingSuccess<u128>, Self::Error> {
+        // Check that we are consuming at least one input
+        ensure!(
+            !input_data.is_empty(),
+            ConstraintCheckerError::SpendingNothing
+        );
+
+        let mut total_input_value: u128 = 0;
+        let mut total_output_value: u128 = 0;
+
+        // Check that sum of input values < output values
+        for input in input_data {
+            let utxo_value = input
+                .extract::<Coin<ID>>()
+                .map_err(|_| ConstraintCheckerError::BadlyTyped)?
+                .0;
+            total_input_value = total_input_value
+                .checked_add(utxo_value)
+                .ok_or(ConstraintCheckerError::ValueOverflow)?;
         }
+
+        for utxo in output_data {
+            let utxo_value = utxo
+                .extract::<Coin<ID>>()
+                .map_err(|_| ConstraintCheckerError::BadlyTyped)?
+                .0;
+            ensure!(utxo_value > 0, ConstraintCheckerError::ZeroValueCoin);
+            total_output_value = total_output_value
+                .checked_add(utxo_value)
+                .ok_or(ConstraintCheckerError::ValueOverflow)?;
+        }
+
+        ensure!(
+            total_output_value <= total_input_value,
+            ConstraintCheckerError::OutputsExceedInputs
+        );
+
+        // Priority is based on how many token are burned
+        // Type stuff is kinda ugly. Maybe division would be better?
```

Comment:
Still a relevant comment?

Reply:
It is still relevant to the code, but it is not introduced in this PR. This code block was already here and git just didn't notice it's the same. I support addressing this, but not in this PR.

Reply:
Addressed in fa797b50f1d8fe70581ea2e29203472c08ca4758